### ISSUE TITLE
Unified_inventory mod upgrade

### ISSUE THREAD

**Issue by Klaranth** [pandorabox-io/in-game#218]
Hedgehog ; BuckarooBanzai ; Wsor
When hitting I for Inventory, the popup screen starts with crafting. Is there a way to make it start with the bags?

Not sure, but it would be nice if the unified_inventory mod would remember on what "page" you were before (crafting, recipe, bags, etc)

It just means an extra packet to the player on close
Since you would need to set a new formspec string every time

**Comment by Klaranth** [comment]
Huhhila
override unified_inventory misc_set_day and misc_set_night to be invisible instead of greyed out (would avoid the mostly-unnecessary scrollbar)

BuckarooBanzai 
The unified inventory has a few edges where it can "improved" right now, the issue from FeXoR with the misaligned icons comes to mind too :/

Hedgehog 
If that ( of FeXoR) should be added to github, do you know where that info is found?

BuckarooBanzai 
Reported on IRC here: https://irc.minetest.net/minetest/2022-04-04#i_5956818 let's see what the idea behind it was
IRC log for #minetest, 2022-04-04
IRC LOG for channel #minetest

wsor 
tldr buckaroo wasnt aware that unified inventory has been formspec v4/min minetest 5.4 for a while now
https://github.com/minetest-mods/unified_inventory/blob/master/mod.conf#L8

BuckarooBanzai
he has been aware but ignored it pretty well ;)

**Comment by Klaranth** [comment]
BuckarooBanzai
some linux distros' dependency managers are notoriously known for their out-of-date packages, mine (xubuntu 21.10) still has minetest/focal 5.1.1+repack-1build1 amd64 as the "latest" :P

**Comment by Klaranth** [comment]
Huhhila
1hit: add from_item to craftdb get_recipes to make it possible to work around unified inventory craft guide being broken (doesn't show what can be crafted from an item like say, any lump - although for coal it shows some dye recipes)

SX 
Revert unified inventory to some previous known good version?

Hedgehog
Can't that be done by pulling an earlier version from the github repo ?

SX 
Yes, that's exactly how it is done if it will be done. There's of course question what'll be lost if reverted and if reverting to older version is worth the trouble. Another option is to live with it until fixed.

Hedgehog
Or fix it yourself and send it upstream? :)

SX 
That's exactly "another option", I just left assignment open (as I'm not going to start working on that now)
Probably will be fixed anyway by someone some day... reverting to old version is more about temporary solution until feature is fixed / finished.
If it (reverting to older) is worth the trouble depends mostly on how bad it is, will ireverting also remove some important features/fixes and if someone is going to work on it soon or not. 
To me it sounds like while bug is announcing to anyone who uses crafting guide it mostly affects new players / player who do not use automated / semi-automated factories for crafting stuff.
Especially new players who do not know recipes.

**Comment by Klaranth** [comment]
wsor
https://github.com/pandorabox-io/pandorabox-mods/pull/2244 should provided backwards compat (to the max of whats possible at least) for all formspecs on the server

Huhhila
other pandorabox mods depend on v4 formspecs too

wsor 
yeah, and depending on which elements they use, depends on how well/to what degree they work on older clients.

Huhhila
anyway I'd say unified inventory changes breaking craft guide forward recipe queries (what-can-I-make-from-this) is worse regression than any of the formspec related ones.

Huhhila
Anyone know whether that particular issue is limited to pandorabox or can be replicated standalone too?

wsor 
this isnt a break, the server was assuming that the formspec would work for all clients when it miniumly supports 5.4 and up

**Comment by Klaranth** [comment]
SX 
crafting guide recipes not showing bug ^^

Huhhila
SX: wondering why it manages to show some black dye recipes for default:coal_lump but none of the others...

SX 
formspec thing was fixed with compatibility layer hack and confirmed to be working? or maybe so far just tested and not yet deployed. on the other hand recipe bug hasnt received any love so far
so it is not completely broken but just some things... I've not looked into that really at all and it is possible that it is some customization conflict. would be good to test with some limited set of basic mods or just mtg to see if it would seem to work then...

**Comment by Klaranth** [comment]
<Huhhila> #218 comments between "work around" and "conflict" were obsoleted by https://github.com/pandorabox-io/pandorabox-mods/commit/4cd620d7545b1483364ed28cdaebc3461bd7c421 (fix for the issue discussed earlier works just fine)

**Comment by Klaranth** [comment]
@OgelGames
unified inventory addition: have a way to filter/search for a particular method of crafting (crafting, cooking, digging, etc), possibly by clicking on the icon that is shown when viewing a recipe

@S-S-X 
More inventory stuff: All Technic recipe types to crafting guide. (guide does not fully support some complex recipes but I've tested it once and I think it is still helpful even if incomplete).

**Comment by OgelGames** [comment]
Almost everything here has been fixed by now, I added my suggestion to an upstream issue: https://github.com/minetest-mods/unified_inventory/issues/233. Closing this.
